### ISSUE TITLE
Missing completions for intersection error types

### ISSUE THREAD

**Issue by TharushiJay** [ballerina-platform/ballerina-lang#34509]
**Description:**
<img width="677" alt="Screenshot 2022-01-05 at 10 32 52" src="https://user-images.githubusercontent.com/27485094/148163163-d325d8ec-a6fa-4af6-9d2e-e70484cd8add.png">

**Steps to reproduce:**
```
type IoError distinct error;

type FileErrorDetail record {
   string filename;
};

type FileIoError IoError & error<FileErrorDetail>;


public function main() {

    FileIo<cursor>
}
```

**Affected Versions:**
SL Beta6

**Comment by nadeeshaan** [comment]
Here the visible symbols does not include the particular type symbol. Also, another issue is that the navigation features such as definitions and references are not working even after we use the type.

eg: 
```
type IoError distinct error;

type FileErrorDetail record {
   string filename;
};

type FileIoError IoError & error<FileErrorDetail>;


public function main() {

    FileI<cursor>oError err;
}
```

Goto definition is not working at the cursor as well.

**Comment by mohanvive** [comment]
Related, https://github.com/ballerina-platform/ballerina-lang/issues/34649

**Comment by pubudu91** [comment]
Blocked by #35068

**Comment by nipunayf** [comment]
The completions are working as expected, along with navigation and references. Added tests to cover this scenario as well: https://github.com/ballerina-platform/ballerina-lang/pull/41882.

<img width="561" alt="image" src="https://github.com/ballerina-platform/ballerina-lang/assets/59343084/b34aaed9-5913-4f4a-a3b6-fa9750587e1b">
